jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -434,29 +466,46 @@ private void setModules(SchemaGeneratorConfigBuilder configBuilder) throws MojoE
      */
     private URLClassLoader getClassLoader() {
         if (this.classLoader == null) {
-            List<String> runtimeClasspathElements = null;
-            try {
-                runtimeClasspathElements = project.getRuntimeClasspathElements();
-            } catch (DependencyResolutionRequiredException e) {
-                this.getLog().error("Failed to resolve runtime classpath elements", e);
-            }
+            List<URL> urls = getUrls(Classpath.WITH_COMPILE_DEPENDENCIES);
+            this.classLoader = new URLClassLoader(urls.toArray(new URL[0]),
+                Thread.currentThread().getContextClassLoader());
+        }
+        return this.classLoader;
+    }
 
-            if (runtimeClasspathElements != null) {
-                URL[] runtimeUrls = new URL[runtimeClasspathElements.size()];
-                for (int i = 0; i < runtimeClasspathElements.size(); i++) {
-                    String element = runtimeClasspathElements.get(i);
-                    try {
-                        runtimeUrls[i] = new File(element).toURI().toURL();
-                    } catch (MalformedURLException e) {
-                        this.getLog().error("Failed to resolve runtime classpath element", e);
-                    }
-                }
-                this.classLoader = new URLClassLoader(runtimeUrls,
-                        Thread.currentThread().getContextClassLoader());
+    private List<URL> getUrls(Classpath classpath) {
+
+        List<String> classPathElements = null;
+        try {
+            switch (classpath) {
+            case WITH_COMPILE_DEPENDENCIES:
+                classPathElements = project.getCompileClasspathElements();
+                break;
+            case WITH_RUNTIME_DEPENDENCIES:
+                classPathElements = project.getRuntimeClasspathElements();
+                break;
+            case PROJECT_ONLY:
+                classPathElements = new ArrayList<>();
+                classPathElements.add(project.getBuild().getOutputDirectory());
+                break;
+            default:
+                throw new IllegalArgumentException("Classpath " + classpath + " not supported");
             }
+        } catch (DependencyResolutionRequiredException e) {
+            this.getLog().error("Failed to resolve classpath elements", e);
+            return Collections.emptyList();
         }
 
-        return this.classLoader;
+        List<URL> urls = new ArrayList<>(classPathElements.size());
+        for (int i = 0; i < classPathElements.size(); i++) {
+            String element = classPathElements.get(i);
```

Comment:
**suggestion (non-blocking)**: no need to do look-up by index anymore, now that we're not re-using the index variable for assigning entries to particular target array positions.
Seems like a left-over from the previous code. 😉
```suggestion
        for (String element : classPathElements) {
```

Reply:
Changed that